src/algorithms/digi/SiliconChargeSharing.cc

```diff
@@ -0,0 +1,209 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+// Copyright (C) 2024-2025 Chun Yuen Tsang, Prithwish Tribedy, Simon Gardner
+//
+// Spread energy deposition from one strip to neighboring strips within sensor boundaries
+
+#include <DD4hep/Alignments.h>
+#include <DD4hep/DetElement.h>
+#include <DD4hep/Handle.h>
+#include <DD4hep/Objects.h>
+#include <DD4hep/Readout.h>
+#include <DD4hep/Segmentations.h>
+#include <DD4hep/Shapes.h>
+#include <DD4hep/VolumeManager.h>
+#include <DD4hep/detail/SegmentationsInterna.h>
+#include <DDSegmentation/MultiSegmentation.h>
+#include <DDSegmentation/Segmentation.h>
+#include <Evaluator/DD4hepUnits.h>
+#include <Math/GenVector/Cartesian3D.h>
+#include <Math/GenVector/DisplacementVector3D.h>
+#include <TGeoBBox.h>
+#include <TGeoMatrix.h>
+#include <algorithms/geo.h>
+#include <edm4hep/Vector3d.h>
+#include <fmt/core.h>
+#include <algorithm>
+#include <cmath>
+#include <gsl/pointers>
+#include <set>
+#include <stdexcept>
+#include <typeinfo>
+#include <utility>
+#include <vector>
+
+#include "DD4hep/Detector.h"
+#include "SiliconChargeSharing.h"
+#include "algorithms/digi/SiliconChargeSharingConfig.h"
+
+namespace eicrecon {
+
+void SiliconChargeSharing::init() {
+  m_converter = algorithms::GeoSvc::instance().cellIDPositionConverter();
+  m_seg       = algorithms::GeoSvc::instance().detector()->readout(m_cfg.readout).segmentation();
+}
+
+void SiliconChargeSharing::process(const SiliconChargeSharing::Input& input,
+                                   const SiliconChargeSharing::Output& output) const {
+  const auto [simhits] = input;
+  auto [sharedHits]    = output;
+
+  for (const auto& hit : *simhits) {
+
+    auto cellID = hit.getCellID();
+
+    auto element = &m_converter->findContext(cellID)->element; // volume context
+    // ToDo: Move this to init() and set it once for every detelement associated with the readout
+    // Set transformation map if it hasn't already been set
+    m_transform_map.try_emplace(element, &element->nominal().worldTransformation());
+    m_segmentation_map.try_emplace(element, getLocalSegmentation(cellID));
+
+    // Try and get a box of the detectorElement solid
+    try {
+      dd4hep::Box box = element->solid();
+
+      if (box) {
+        m_x_range_map.try_emplace(element, box->GetDX());
+        m_y_range_map.try_emplace(element, box->GetDY());
```

Comment:
Can/should the map be from element to pair {x,y}?

Reply:
And should there not be a condition that this is CartesianGridXY, since for CartesianGridXZ the ranges would be incorrect?

Reply:
I have swapped to a {x,y} pair.
I believe the getLocalSegmentation should currently restrict it to XY segmentation, this could be generalized if necessary in the future.